features/lesson/src/main/kotlin/com/teacherworkout/features/lesson/LessonStartScreen.kt

```diff
@@ -0,0 +1,242 @@
+                    LessonThemeTitleText(title = lesson.lessonThemeTitle)
+                    Spacer(modifier = Modifier.height(space16dp))
+                    //TODO: determine which lesson attribute to use for the duration text
+                    DurationText(duration = lesson.remainingMinutes.toString() + " min")
+                    Spacer(modifier = Modifier.height(space24dp))
+                    StartContinueButton(
+                        lessonStarted = lesson.started,
+                        onClick = { onSendEvent(LessonContract.Event.StartContinue()) }
+                    )
+                    Spacer(modifier = Modifier.height(space24dp))
+                    if (lesson.saved) {
+                        UnsaveButton(onClick = { onSendEvent(LessonContract.Event.Unsave()) })
+                    } else {
+                        SaveButton(onClick = { onSendEvent(LessonContract.Event.Save()) })
+                    }
+                }
+            }
+        }
+    }
+}
+
+@Composable
+private fun StartContinueButton(
+    lessonStarted: Boolean,
+    onClick: () -> Unit
+) {
+    val minTouchSize = dimensionResource(id = R.dimen.min_touch_size)
+
+    Button(
+        modifier = Modifier
+            .fillMaxWidth()
+            .height(minTouchSize),
+        onClick = onClick,
+        shape = RoundedCornerShape(50)
+    ) {
+        Text(stringResource(if (lessonStarted) R.string.start_lesson_label else R.string.continue_lesson_label))
+    }
+}
+
+@Composable
+private fun LessonTitleText(title: String) {
+    Text(
+        modifier = Modifier.fillMaxWidth(),
+        text = title,
+        textAlign = TextAlign.Center,
+        style = TextStyle(
+            color = MaterialTheme.colors.primary,
+            fontSize = MaterialTheme.typography.h4.fontSize,
+            fontWeight = FontWeight.Bold
+        )
+    )
+}
+
+@Composable
+private fun LessonThemeTitleText(title: String) {
+    Text(
+        modifier = Modifier.fillMaxWidth(),
+        text = title,
+        textAlign = TextAlign.Center,
+        color = MaterialTheme.colors.primary,
+    )
+}
+
+@Composable
+private fun DurationText(duration: String) {
+    Row(
+        modifier = Modifier.fillMaxWidth(),
+        horizontalArrangement = Arrangement.Center
+    ) {
+        //TODO: set the content description to the duration icon
+        Icon(
+            imageVector = Icons.Default.Schedule,
+            contentDescription = "",
+            tint = MaterialTheme.colors.primary
+        )
+        Text(
+            text = " ".plus(duration),
+            color = MaterialTheme.colors.primary
+        )
+    }
+}
+
+@Composable
+private fun SaveButton(onClick: () -> Unit) {
+    val minTouchSize = dimensionResource(id = R.dimen.min_touch_size)
+
+    OutlinedButton(
+        modifier = Modifier
+            .fillMaxWidth()
+            .height(minTouchSize),
+        onClick = onClick,
+        shape = RoundedCornerShape(50),
```

Comment:
Maybe we can extract this value in a variable as it seems to be used in multiple places?

Reply:
Maybe we could leave this for a different PR? I think we should create a common button with that shape to be used across multiple screens.